File: server/bean/Room.js
/*
 * @File:   Room.js
 * @Author: yintaibing, SCUT Software Engineering, Grade 2012, Class 6.
 * @Date:   2016/04/15
 */

var Util = require('util');
var BaseBean = require('./BaseBean.js');

function Room(id, title, publisherId) {
	BaseBean.call(this, id);

	this.title = title;
	this.publisherId = publisherId;
	this.categoryId = null;
	this.isLiving = false;
}

Util.inherits(Room, BaseBean);

Room.prototype.getTitle = function() {
	return this.title;
};

Room.prototype.setTitle = function(title) {
	this.title = title;
};

Room.prototype.getPublisherId = function() {
	return this.publisherId;
};

Room.prototype.setPublisherId = function(publisherId) {
	this.publisherId = publisherId;
};

Room.prototype.getCategoryId = function() {
	return this.categoryId;
};

Room.prototype.setCategoryId = function(categoryId) {
	this.categoryId = categoryId;
};

Room.prototype.getIsLiving = function() {
	return this.isLiving;
};

Room.prototype.setIsLiving = function(isLiving) {
	this.isLiving = isLiving;
};

module.exports = Room;

File: client-android/TurtleTV/src/com/turtletv/android/bean/Room.java
package com.turtletv.android.bean;

public class Room extends BaseBean {
	public static final String COL_TITLE = "title";
	public static final String COL_PUBLISHER_ID = "publisherId";
	public static final String COL_CATEGORY_ID = "categoryId";
	public static final String COL_IS_LIVING = "isLiving";
	public static final String COL_PUBLISHER_NAME = "publisherName";
	public static final String COL_AUDIENCE_NUM = "audienceNum";
	
	private String title;
	private int publisherId;
	private int categoryId;
	private boolean isLiving;
	
	private String publisherName;
	private int audienceNum;
	
	public String getPublisherName() {
		return publisherName;
	}

	public void setPublisherName(String publisherName) {
		this.publisherName = publisherName;
	}

	public int getAudienceNum() {
		return audienceNum;
	}

	public void setAudienceNum(int audienceNum) {
		this.audienceNum = audienceNum;
	}

	public String getTitle() {
		return title;
	}
	
	public void setTitle(String title) {
		this.title = title;
	}
	
	public int getPublisherId() {
		return publisherId;
	}
	
	public void setPublisherId(int publisherId) {
		this.publisherId = publisherId;
	}
	
	public int getCategoryId() {
		return categoryId;
	}
	
	public void setCategoryId(int categoryId) {
		this.categoryId = categoryId;
	}
	
	public boolean getIsLiving() {
		return isLiving;
	}
	
	public void setIsLiving(boolean isLiving) {
		this.isLiving = isLiving;
	}
}
